---
title: "deseq2_snai1_ko"
output: html_document
date: "2025-04-15"
---

# RNAseq Differential Expression for Snai1 KO in Breast Cancer Cells
Reference: https://www.ebi.ac.uk/gxa/experiments/E-MTAB-5244/Results

## Import Libraries
```{r}
if(!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install("DESeq2")
library(DESeq2)
```
```{r}
BiocManager::install("biomaRt")
BiocManager::install("EnhancedVolcano")
```

```{r}
# Load libraries
library(ggplot2)
library(EnhancedVolcano)
library(biomaRt)
```

## Download data
```{r}
# Download raw counts
counts <- read.delim("E-MTAB-5244-raw-counts.tsv")
head(counts)

# Download metadata
metadata <- read.delim("E-MTAB-5244-experiment-design.tsv")
head(metadata)
```
## Data Wrangling
DESeq expects the counts to have gene IDs as row names and not numbers
```{r}
rownames(counts) <- counts$Gene.ID
head(counts)

# Remove unused columns (gene ID and gene name)
genes = counts[, c("Gene.ID", "Gene.Name")]
counts = counts[, -c(1,2)]
head(counts)
```
```{r}
# Change row names in metadata to sample IDs
rownames(metadata) <- metadata$Run
head(metadata)

# Only keep columns of interest (genotype wildtype vs knockout )
metadata <- metadata[, c("Sample.Characteristic.genotype."), drop=FALSE]
colnames(metadata) <- c("genotype")
metadata
```

```{r}
# Remove spaces in names to avoid DESeq warnings
metadata$genotype[metadata$genotype == "wild type genotype"] = 'wildtype'
metadata$genotype[metadata$genotype == "Snai1 knockout"] = 'knockout'

# Turn genotype into a factor
metadata$genotype <- factor(metadata$genotype, levels = c("wildtype", "knockout")) # wildtype is the baseline (place 1st)
metadata
```

## Spot check expression for knockout gene SNAI1
```{r}
# Verify that SNAI1 is indeed knocked out
gene_id <- genes$Gene.ID[genes$Gene.Name == 'SNAI1']
gene_counts <- counts[gene_id, ]
gene_counts
```

```{r}
# Combine the metadata column to see if the knockout samples are the ones with low expression
gene_data <- cbind(metadata, counts=as.numeric(gene_counts))
gene_data
```
Results makes sense: knockout SNAI1 is lower than wildtype

```{r}
# Visualize SNAI1 expression for wildtype vs knockout
ggplot(gene_data, aes(x = genotype, y = counts, fill = genotype)) + geom_boxplot()
```

## Run DESeq
```{r}
dds <- DESeqDataSetFromMatrix(countData = counts, colData = metadata, design = ~genotype)

# Ignore genes with low counts
dds <- dds[rowSums(counts(dds)) > 10, ]

# Run DESeq
dds <- DESeq(dds)

# Compare expression
res <- results(dds, contrast = c("genotype", "knockout", "wildtype"), alpha = 1e-5)
res
```
```{r}
# Merge gene name into data frame so can compare using gene name
res_df <- as.data.frame(res)
head(res_df)

res_df <- merge(res_df, genes, by = 'row.names')
head(res_df)
```

```{r}
genes_to_check <- c("THY1", "PASD1")
res_df[res_df$Gene.Name %in% genes_to_check, ]
```

## Visualization

```{r}
plotMA(res)
```
Volcano Plot
```{r}
EnhancedVolcano(res, lab=rownames(res), x = "log2FoldChange", y =  "pvalue")
```
